.github/workflows/qt-ci.yml

```diff
@@ -233,6 +233,14 @@ jobs:
           key: Qt_${{ matrix.name }}_${{ matrix.qt_version }}
           max-size: 200M
 
+      - name: npm install
+        run: npm install --ignore-scripts
+        working-directory: ./
+
+      - name: Generate shaders
+        working-directory: ./
+        run: node shaders/generate_shader_code.js --root ./ --opengl --metal --strip --compress --amalgamate
```

Comment:
I'm on holidays so just seeing a Qt comment on my phone. Is this for testing or is it a build dependency? I would really like to avoid npm dependence for Qt.

Reply:
This is just for CI to try and validate shaders are kept up to date with PRs that change them. When changes to shaders are made, this tool is run to re-generate them and changes are (supposed to be) submitted with the PR. During normal builds, you shouldn't need to run this tool.

This is also run on CI for the Linux platform so Bloaty can see the compressed version of shaders, while we only check in the human readable versions.

Reply:
Ok, thanks for clarification.